platform/osx/os_osx.mm

```diff
@@ -408,25 +405,31 @@ - (void)windowDidBecomeKey:(NSNotification *)notification {
 
 		OS_OSX::singleton->get_main_loop()->notification(MainLoop::NOTIFICATION_WM_FOCUS_IN);
 	}
+
+	window_focused = true;
```

Comment:
Should be `OS_OSX::singleton->window_focused = true`, these notifications aren't part of `OS_OSX` class.